models/core-java/src/main/java/org/commonjava/indy/model/core/AccessChannel.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (C) 2011 Red Hat, Inc. (jdcasey@commonjava.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.model.core;
+
+/**
+ * Enumeration to distinguish between different access channels to stores.
+ *
+ * @author pkocandr
+ */
+public enum AccessChannel
```

Comment:
I also thought about string constants, but for clients it is more clear to have an enum. Also it is much easier to find all allowed values like this rather than when it's spread across multiple constants and one has to scan all occurrences of the EventMetadata constructor and looking for the values and their meaning that gets filled in.
I agree that once we refactor the Maven specific stuff out of core we will also have to alter the values that gets filled in. Not sure if that will mean that we also provide some separate channel for maven, like /api/maven/<store type>/<store name>, or if we use the current channel and will get harder to distinguish between Maven and others.